runners/flink/src/main/java/org/apache/beam/runners/flink/metrics/FlinkMetricContainer.java

```diff
@@ -62,22 +67,27 @@
   private final Map<String, FlinkGauge> flinkGaugeCache;
   private final MetricsAccumulator metricsAccumulator;
 
-  public FlinkMetricContainer(RuntimeContext runtimeContext) {
+  public FlinkMetricContainer(RuntimeContext runtimeContext, boolean accumulatorDisabled) {
     this.runtimeContext = runtimeContext;
     this.flinkCounterCache = new HashMap<>();
     this.flinkDistributionGaugeCache = new HashMap<>();
     this.flinkGaugeCache = new HashMap<>();
 
-    Accumulator<MetricsContainerStepMap, MetricsContainerStepMap> metricsAccumulator =
-        runtimeContext.getAccumulator(ACCUMULATOR_NAME);
-    if (metricsAccumulator == null) {
+    Accumulator<MetricsContainerStepMap, MetricsContainerStepMap> metricsAccumulator;
+    if (accumulatorDisabled) {
+      // Do not register the accumulator with Flink
       metricsAccumulator = new MetricsAccumulator();
```

Comment:
Is this accumulator still doing anything if it's not registered?

Reply:
Yes, it holds all local metrics.